File: designacoes/src/main/java/com/example/designacoes/domains/Relatorio.java
package com.example.designacoes.domains;

import lombok.Data;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;


@Entity
@Data
public class Relatorio {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "id_publicador")
    public Publicador publicador;

    @Column
    private Long mes;

    @Column
    private Long ano;

    @Column
    private Boolean pioneiroAuxiliar;

    @Column
    private Long publicacoes;

    @Column
    private Long videos;

    @Column
    private Long horas;

    @Column
    private Long revisitas;

    @Column
    private Long estudos;

    @Column
    private String observacao;
}

File: designacoes-front/src/app/shared/domain/relatorio.ts
export class Relatorio {
    nomePublicador : string = '';
    id : number = 0;
    idPublicador : number = 0;
    mes : number = 0;
    ano : number = 0;
    pioneiroAuxiliar: boolean = false;  
    publicacoes : number = 0;
    videos : number = 0;
    horas : number = 0;
    revisitas : number = 0;
    estudos : number = 0;
    observacao: string = "";  
 
    constructor(nomePublicador: string, mes: number, ano: number, pioneiroAuxiliar: boolean, publicacoes: number, videos: number, horas: number, revisitas: number, estudos: number, observacao: string) {
       this.nomePublicador = nomePublicador;
       this.mes = mes;
       this.ano = ano;
       this.pioneiroAuxiliar = pioneiroAuxiliar;
       this.publicacoes = publicacoes;
       this.videos = videos;
       this.horas = horas;
       this.revisitas = revisitas;
       this.estudos = estudos;
       this.observacao = observacao;
     }
 } 

  